.github/workflows/release.yml

```diff
@@ -23,6 +23,16 @@ jobs:
           fetch-depth: 0
           submodules: recursive
 
+      - name: Setup .NET SDK
+        uses: actions/setup-dotnet@v4
+        env:
+          NUGET_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          dotnet-version: |
+            6.0.x
+            7.0.x
+          source-url: https://nuget.pkg.github.com/open-feature/index.json
```

Comment:
Nope, it should not, because in that step we explicitly pass the requisite API token.

This pattern here is just telling `actions/setup-dotnet` to (more or less) run `dotnet nuget update source ...` for this `source-url` using the ambient `NUGET_AUTH_TOKEN` while setting up the SDK.

As an alternative to pre-configuring this `source-url` here, we could instead pass additional sources and tokens to `dotnet restore`, but that gets really messy really fast, so the usual pattern is to just pre-configure your authenticated NuGet sources as part of setting up the .NET SDK.

I get this question a lot in repos where I'm setting up GitHub Actions for the first time, so often find myself wishing they had instead implemented `source-url` as `source-urls: [ ... ]` which would let us be more expressive about `NUGET_AUTH_TOKEN` being for, and only for, one `source-url`, but as things stand repos that want to pre-wire two auth'd feeds just end up running this step twice, once for each `source-url` and `NUGET_AUTH_TOKEN`.

Here's a snippet from the CI logs that kind of allude to this^ story (albeit, not very clearly, but if you squint you can kind of picture what its doing):

![image](https://github.com/open-feature/dotnet-sdk-contrib/assets/21338699/5a009a92-7cf5-4df3-8386-135032b60959)